compiler/ballerina-lang/src/main/java/io/ballerina/projects/JBallerinaBackend.java

```diff
@@ -267,7 +267,8 @@ public void performCodeGen(ModuleContext moduleContext, CompilationCache compila
         }
 
         String testJarFileName = jarFileName + TEST_JAR_FILE_NAME_SUFFIX;
-        CompiledJarFile compiledTestJarFile = jvmCodeGenerator.generateTestModule(bLangPackage.testablePkgs.get(0));
+        CompiledJarFile compiledTestJarFile = jvmCodeGenerator.generateTestModule(bLangPackage.testablePkgs.get(0),
+                                                                                  compiledJarFile.getbStringVarMap());
```

Comment:
I have added a separate class for generating string constants only defined in test modules. Those will be part of test module. If module string constant is used by the test module, those constants will be replaced by module constants. Basicall we have two maps now.